src/Microsoft.OData.Client/Serialization/ODataWriterWrapper.cs

```diff
@@ -47,6 +48,41 @@ internal static ODataWriterWrapper CreateForEntry(ODataMessageWriter messageWrit
             return new ODataWriterWrapper(messageWriter.CreateODataResourceWriter(), requestPipeline);
         }
 
+        /// <summary>
+        /// Creates the OData delta feed writer.
+        /// </summary>
+        /// <param name="messageWriter">The message writer.</param>
+        /// <param name="entitySetName">The EntitySetName of the top level descriptor</param>
+        /// <param name="requestPipeline">The request pipeline configuration.</param>
+        /// <param name="messageWrapper">The messagewrapper</param>
+        /// <param name="model">The EdmModel.</param>
+        /// <returns>The odata Writer Wrapper</returns>
+        internal static ODataWriterWrapper CreateForDeltaFeed(ODataMessageWriter messageWriter, string entitySetName, DataServiceClientRequestPipelineConfiguration requestPipeline, ODataRequestMessageWrapper messageWrapper, RequestInfo requestInfo)
+        {
+            EntityDescriptor entityDescriptor = messageWrapper.Descriptor as EntityDescriptor;
+            IEdmEntityContainer entityContainer = requestInfo.Format.ServiceModel.EntityContainer;
+            IEdmStructuredType entityType = requestInfo.Model.GetOrCreateEdmType(entityDescriptor.Entity.GetType()) as IEdmStructuredType;
+            string setName;
+
+            if (string.IsNullOrEmpty(entityDescriptor.EntitySetName))
+            {
+                setName = entitySetName;
+            }

```

Comment:
So, if the user explicitly specifies an entity set name, it may still be overridden by the one on the entity descriptor?

Reply:
How will they explicitly specify it? If they explicitly specify it then it will on the `entityDescriptor`.